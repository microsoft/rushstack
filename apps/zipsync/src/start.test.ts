// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { execSync } from 'child_process';

describe('CLI Tool Tests', () => {
  it('should display help for "zipsync --help"', () => {
    const startOutput = execSync('node lib/start.js --help').toString();
    expect(startOutput).toMatchSnapshot();
  });
});
