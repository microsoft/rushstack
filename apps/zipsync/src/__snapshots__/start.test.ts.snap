// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Tool Tests should display help for "zipsync --help" 1`] = `
"
zipsync 0.0.0 - https://rushstack.io

usage: zipsync [-h] [-d] [-v] -m {pack,unpack} -a ARCHIVE_PATH -t
               TARGET_DIRECTORIES -b BASE_DIR -z {store,deflate,zstd,auto}
               

Optional arguments:
  -h, --help            Show this help message and exit.
  -d, --debug           Show the full call stack if an error occurs while 
                        executing the tool
  -v, --verbose         Show verbose output
  -m {pack,unpack}, --mode {pack,unpack}
                        The mode of operation: \\"pack\\" to create a zip archive,
                         or \\"unpack\\" to extract files from a zip archive
  -a ARCHIVE_PATH, --archive-path ARCHIVE_PATH
                        Zip file path
  -t TARGET_DIRECTORIES, --target-directory TARGET_DIRECTORIES
                        Target directories to pack or unpack
  -b BASE_DIR, --base-dir BASE_DIR
                        Base directory for relative paths within the archive
  -z {store,deflate,zstd,auto}, --compression {store,deflate,zstd,auto}
                        Compression strategy when packing. \\"deflate\\" and 
                        \\"zlib\\" attempts compression for every file (keeps 
                        only if smaller); \\"auto\\" first skips 
                        likely-compressed types before attempting \\"deflate\\" 
                        compression; \\"store\\" disables compression.

[1mFor detailed help about a specific command, use: zipsync <command> -h[22m
"
`;
